test/test_memmap.py

```diff
@@ -137,9 +150,38 @@ def test_memmap_clone():
     assert m2c == m1
 
 
-def test_memmap_tensor():
-    t = torch.tensor([[1, 2, 3], [4, 5, 6]])
-    assert (torch.tensor(t) == t).all()
+@pytest.mark.parametrize("device", get_available_devices())
+def test_memmap_same_device_as_tensor(device):
+    """
+    Created MemmapTensor should be on the same device as the input tensor.
+    Check if device is correct when .to(device) is called.
+    """
+    t = torch.tensor([1], device=device)
+    m = MemmapTensor(t)
+    assert m.device == torch.device(device)
+    for other_device in get_available_devices():
+        m.to(other_device)
+        assert m.device == torch.device(other_device)
```

Comment:
Can we test a simple operation, like `assert (m + torch.ones(3, 4, device= other_device) == 1).all()`?
It should return an error if something is not on the same device

Reply:
Added test cases.